### PR TITLE
internal: vendor lsp-server

### PR DESCRIPTION
**Pull request by matklad**

### CODE REVIEW

File: lib/la-arena/Cargo.toml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name = "la-arena"
 version = "0.3.0"
 description = "Simple index-based arena without deletion."
 license = "MIT OR Apache-2.0"
-repository = "https://github.com/rust-lang/rust-analyzer"
+repository = "https://github.com/rust-lang/rust-analyzertree/master/lib/la-arena"
```
Comment (Veykril) [inline lib/la-arena/Cargo.toml:6]:
```suggestion
repository = "https://github.com/rust-lang/rust-analyzer/tree/master/lib/la-arena"
```

**Comment by matklad** [comment]
I didn't try to transplant the history, as the crate seems simple enough that there's probably little value in it.

**Comment by MartinNowak** [comment]
> I didn't try to transplant the history, as the crate seems simple enough that there's probably little value in it.

It's a lot simpler with [subtree merging](https://git-scm.com/book/en/v2/Git-Tools-Advanced-Merging#_subtree_merge) nowadays.
`git merge remotes/lsp-server/main -Xsubtree=lib/lsp-server`

**Pull request merged**